### PR TITLE
Use PrismicImage on Events/ExhibitionsPromo

### PR DESCRIPTION
**Pull request by jamesgorrie**
Waiting on #7371 

Starts using the `PrismicImage`, which uses the [`next/image`](https://nextjs.org/docs/api-reference/next/image).

The main inspiration to do this is to allow me to continue with my Prismic refactoring (#7363). By not using the severly coupled, and impossible to refactor `UiImage` component, this will allow me to move closer to using generic Prismic types over our custom `ParsedFromPrismic` types as it takes a generic Prismic type as an argument.

As mention, we have also hit a stale mate with our Image types and components, where they've proliferated out somewhat (`UIImage`, `Image`, `TASL`, `Picture` etc). This will hope to move in the direction of having a simpler model on the frontend.

I have also tried to use a more structured API when using [`sizes`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-sizes). i.e.
```ts
{
  sizes: {
    breakpointName: ratio
  }
}

// e.g. we have 3 images in a row on xlarge breakpoint
{
  sizes: {
    xlarge: 1/3
  }
}

// Currently we just use the string
// (min-width: 1420px) 386px
```

For those not faniliar with the sizes/srcset APIs - this is what is produced from CardGrid that breaks from 3 to 2 to 1 as we size down. `(min-width: x) Ypx` means at `x` breakpoint, use closest to `Y` from the `srcset`, which is a combined list from [these values](https://github.com/wellcomecollection/wellcomecollection.org/pull/7372/files#diff-f51c3d1edd5c339bb6a1f5d39554b041cd7d2358c71278e36b056973c4a395ceR8-R9).

```TSx
<UiImage sizes="(min-width: 1420px) 386px, (min-width: 960px) calc(28.64vw - 15px), (min-width: 600px) calc(50vw - 54px), calc(100vw - 36px)" />


<PrismicImage sizes="(min-width: 1338px) 446px, (min-width: 960px) 33.33333333333333vw, (min-width: 600px) 50vw, (min-width: 0px) 100vw" />
```

I have compiled some results on the [On Happiness](https://wellcomecollection.org/seasons/YEY3ZBAAACEASBjA)
with pages with articles, events and exhibitions in the `CardGrid`.

```
UiImage
1338+
initial: 268.38 kB / 278.53 kB transferred
fullscroll: 334.08 kB / 346.27 kB transferred

960
initial: 97.44 kB / 108.16 kB transferred
fullscroll: 133.48 kB / 146.80 kB transferred

600
initial: 55.16 kB / 64.85 kB transferred
fullscroll: 112.02 kB / 125.34 kB transferred

----

PrismicImage
1338+
initial: 78.92 kB / 80.27 kB transferred
fullscroll: 349.65 kB / 356.66 kB transferred

960
initial: 69.61 kB / 73.10 kB transferred
fullscroll: 148.15 kB / 155.19 kB transferred

600
initial: 44.22 kB / 47.20 kB transferred
fullscroll: 126.72 kB / 133.75 kB transferred
   
```

There are small bumps in full scrolled, but a significant saving on initial load. I think this might be because we account for the minutia of padding as the strings generated using a tool to get the exact amount. I feel the understandability of using a more structured API will allow us to be able to move away from current layouts more easily, but also optimise more easily too, so worth it.

### CODE REVIEW

**Comment by alexwlchan** [comment]
I'm going to wait until #7371 is merged before I review this.

**Comment by jamesgorrie** [comment]
@alexwlchan - merged, sorry just had the description writ up, so put it in.

**Comment by jamesgorrie** [comment]
Converting to draft as I've thought of a better way to introduce this.

**Comment by jamesgorrie** [comment]
@wellcomecollection/js-ts-reviewers I've refactored the refactor to:

* Create the new exhibition `PrismicDocument`
* Attach the prismic document to the exhibition model. This would convcern me with data transfer sizes, but we never do this async.
* Pass that document to the `ExhibitionPromo`
* Use the new `PrismicImage` in that promo
* ☝️ the refactoring to the image component explained in the description

This allows me to keep creating and using the new prismic stuff, which will then allow me to remove the old types and parsers.

File: content/webapp/components/ExhibitionPromo/ExhibitionPromo.tsx
Comment (jamesgorrie) [inline content/webapp/components/ExhibitionPromo/ExhibitionPromo.tsx:44]: There's deffinitely some abstration to be done here, as and when though...

File: content/webapp/services/prismic/types.ts
Comment (alexwlchan) [inline content/webapp/services/prismic/types.ts:30]: iot's?

**Review by alexwlchan — APPROVED** [review]
Probably could benefit from another pair of more experienced eyes, but looks okay to me.

**Review by gestchild — APPROVED** [review]
Nice. Making use of next/image and eventually ditching lazysizes has been on my todo list.

**Pull request merged**
